data-prepper-plugins/opensearch-api-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearchapi/OpenSearchAPISource.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.opensearchapi;
+
+import org.opensearch.dataprepper.http.BaseHttpService;
+import org.opensearch.dataprepper.http.BaseHttpSource;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.configuration.PipelineDescription;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.plugin.PluginFactory;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.Source;
+import org.slf4j.LoggerFactory;
+
+@DataPrepperPlugin(name = "opensearch_api", pluginType = Source.class, pluginConfigurationType = OpenSearchAPISourceConfig.class)
+public class OpenSearchAPISource extends BaseHttpSource<Record<Event>> {
+    private static final String SOURCE_NAME = "OpenSearch API";
+    private static final String HTTP_HEALTH_CHECK_PATH = "/";
```

Comment:
Does the health check return any data?

What I meant by my comment from the other PR is that we should return the same JSON format that the OpenSearch root endpoint returns.

e.g.

```
docker run -p 9200:9200 -p 9600:9600 -e "discovery.type=single-node" -d opensearchproject/opensearch:2.11.1
```

```
curl -k -u "admin:admin" 'https://localhost:9200/'
{
  "name" : "9c3406b376ef",
  "cluster_name" : "docker-cluster",
  "cluster_uuid" : "ud9QWcTcQruIwlEIjIpYdg",
  "version" : {
    "distribution" : "opensearch",
    "number" : "2.11.1",
    "build_type" : "tar",
    "build_hash" : "6b1986e964d440be9137eba1413015c31c5a7752",
    "build_date" : "2023-11-29T21:45:35.524809067Z",
    "build_snapshot" : false,
    "lucene_version" : "9.7.0",
    "minimum_wire_compatibility_version" : "7.10.0",
    "minimum_index_compatibility_version" : "7.0.0"
  },
  "tagline" : "The OpenSearch Project: https://opensearch.org/"
}
```

Maybe we don't return all of that. But, maybe we return `{}` to start. What I want to avoid is making this return a health response that will not match with the root endpoint going forward.

I'm ok with just removing the health endpoint for the sake of this PR if you want.

Reply:
Thanks @dlvenable. It makes sense to not return a response which does not match the root endpoint response to opensearch. I have disabled it. If customers want to access, data prepper will respond with a 404.